cdap-ui/app/hydrator/controllers/create/leftpanel-ctrl.js

```diff
@@ -162,7 +162,9 @@ class HydratorPlusPlusLeftPanelCtrl {
       let connections = [];
       let i;
       for (i=0; i<nodes.length - 1 ; i++) {
-        connections.push({ from: nodes[i].name, to: nodes[i+1].name });
+        if(nodes[i] && nodes[i + 1]) {
+          connections.push({ from: nodes[i].name, to: nodes[i+1].name });
```

Comment:
When we dont have source, transforms or sinks present in config, undefined value gets added to nodes array so it needs this condition